src/test/test-unit-serialize.c

```diff
@@ -0,0 +1,51 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "service.h"
+#include "tests.h"
+
+#define EXEC_START_ABSOLUTE \
+        "ExecStart 0 /bin/sh \"sh\" \"-e\" \"-x\" \"-c\" \"systemctl --state=failed --no-legend --no-pager >/failed ; systemctl daemon-reload ; echo OK >/testok\""
+#define EXEC_START_RELATIVE \
+        "ExecStart 0 sh \"sh\" \"-e\" \"-x\" \"-c\" \"systemctl --state=failed --no-legend --no-pager >/failed ; systemctl daemon-reload ; echo OK >/testok\""
+
+static void test_deserialize_exec_command_one(const char *key, const char *line, int expected) {
+        int r;
+
+        /* Only vestigial state is created here that is sufficient for log_unit_*() not to crash. */
+        Manager m = {};
```

Comment:
urks, this is ugly.

I'd feel more comfortable if we'd add a macro in manager.h that contains the structured initialization parameter for manager, and that is then both used here and in manager_new(). i.e. the fact that all fd fields here are initialized to 0 (i.e. stdin) makes me feel strange.

Reply:
Indeed. We have `manager_new(MANAGER_TEST_RUN_MINIMAL)` and `unit_new_for_name()`, so I reused that.